var/spack/repos/builtin/packages/migraphx/package.py

```diff
@@ -134,6 +145,12 @@ def cmake_args(self):
             args.append(
                 self.define("GPU_TARGETS", "gfx906;gfx908;gfx90a;gfx1030;gfx1100;gfx1101;gfx1102")
             )
+        if self.spec.satisfies("@6.1:") and self.spec.satisfies("+asan"):
+            args.append(
+                self.define(
+                    "CMAKE_CXX_FLAGS", "-fsanitize=address -shared-libasan -I{0}".format(abspath)
```

Comment:
this is added thru  env.set("CXXFLAGS", "-fsanitize=address -shared-libasan") ? and abspath is added from 5.5.0 onwards L140.

Reply:
L140 overides the CXX_FLAGS previously set by env.set("CXXFLAGS", "-fsanitize=address -shared-libasan"), this is why the flags need to be set again.